src/ast/ast_common.hpp

```diff
@@ -1496,6 +1521,40 @@ struct Ast: public std::enable_shared_from_this<Ast> {
     virtual bool is_ontology_statement() {
         return false;
     }
+
+    /**
+     *\brief Parent getter
+     *
+     * returning a raw pointer may create less problems that the
+     * shared_from_this from the parent.
+     *
+     * \ref Check Ast::parent for more information
+     */
+    virtual Ast* get_parent() const {
+        return parent;
+    }
+
+    /**
+     *\brief Parent setter
+     *
+     * Usually, the parent parent pointer cannot be set in the constructor
+     * because children are generally build BEFORE the parent. Conversely,
+     * we set children parents directly in the parent constructor using
+     * set_parent_in_children()
+     *
+     * \ref Check Ast::parent for more information
+     */
+    virtual void set_parent(Ast* p) {
+        parent = p;
+    }
+
+    /**
+     *\brief Set this object as parent for all the children
+     *
+     * This should be called in every object (with children) constructor
+     * to set the parents.
+     */
+    virtual void set_parent_in_children() {}
```

Comment:
@cattabiani : I merged this already but now noticed this: `set_parent_in_children` should thrown an exception as default method rather than empty?

Reply:
I made this change in  #290: https://github.com/BlueBrain/nmodl/blob/563b9e9034ac38e7f729758ea0fc4144e5304789/src/language/templates/ast/ast.cpp#L73